File: hw2/Codes/Oztunc_Zeynep.rb
#ruby 2.5.1 

#example of unconditional exit using break
loopVar=10

while loopVar >= 1
   loopVar = loopVar - 1
   break
end
puts "The value of the loop variable after the loop is: #{loopVar}"

#example of unconditional exit using next
loopVar=10

while loopVar >= 1
    loopVar = loopVar - 1
    next
end
puts "The value of the loop variable after the loop is: #{loopVar}"

#example of using break in enclosing loops
#outer loop
for i in 0..4 do
     
    puts "Inside outer loop, i is: #{i}"
    puts "------------------------------"
        
    #inner loop
    for j in 6..8 do
        if i==2
            puts "Exiting \n"
            break
        end
   
        puts "Inside inner loop, j is: #{j}"
    end         
    puts "\n"
      
end
    
    
puts "\n Using next \n"
#example of using next in enclosing loops
#outer loop
for i in 0..4 do
     
    puts "Inside outer loop, i is: #{i}"
    puts "------------------------------"
        
    #inner loop
    for j in 6..8 do
        if i==2
            puts "Continuing \n"
            next
        end
        puts "Inside inner loop, j is: #{j}"
    end
        
        
    puts "\n"      
end

File: hw2/Codes/Oztunc_Zeynep.py
#Python 2.7.17

#example of an unconditional exit using break
loopVar = 10
while loopVar >= 1:
  loopVar=loopVar-1
  break
print"Value of the loopVar after the loop is: ",loopVar

print"\n"

#example of an unconditional exit using continue
loopVar = 10
while loopVar >= 1:
    loopVar=loopVar-1

    if loopVar==5: 
        continue
        
print"Value of the loopVar after the loop is: ",loopVar

#example of break statement in enclosing loops
#outer loop
for i in range(0,6):
    print"Inside the outer loop, i:", i
    print"------------------------------"
     
    #inner loop  
    for j  in range(6,9):
        
        #this break statement quits the inner loop
        if i == 2:
            print "Exiting from inner"
            break
        print"Inside the inner loop, j:", j

    print"\n"
print "**********************************************"

#example of continue statement in enclosing loops
#outer loop
for i in range(0,6):
    print"Inside the outer loop, i:", i
    print"------------------------------"
     
    #inner loop  
    for j  in range(6,9):
        
        #this continue statement continues to the next iteration
        if i == 2:
            print "Continuing to the next iteration"
            continue
        print"Inside the inner loop, j:", j

    print"\n"
  
 

